src/main/scala/cognite/spark/v1/PushdownUtilities.scala

```diff
@@ -395,6 +396,28 @@ trait UpdateSchema {
   val updateSchema: StructType
 }
 
-abstract class DeleteSchema {
-  val deleteSchema: StructType = StructType(Seq(StructField("id", DataTypes.LongType)))
+trait UpdateSchemaFromUpsertSchema extends UpsertSchema with UpdateSchema {
+  override val updateSchema: StructType = upsertSchema
 }
+
+trait ReadSchema {
+  val readSchema: StructType
+}
+
+trait DeleteSchema {
+  val deleteSchema: StructType
+}
+
+trait DeleteWithIdSchema extends DeleteSchema {
+  override val deleteSchema: StructType = StructType(Seq(StructField("id", DataTypes.LongType)))
```

Comment:
Would make sense to extract a DeleteByInternalId/DeleteById class too to be more uniform, even if currently delete() code in some places uses DeleteItemByCogniteId union type
Maybe reuse DeleteItem class here, or in some other way make traits here use same pattern

Full move of id/externalId/eitherOfThem to a single deduplicated set of classes in datasource can be a separate PR though

Reply:
Kind of agree, but as you say it complexifies the PR a bit and should probably be separate. The other schema was more straightfoward they were all using externalId only